lib/kernel/src/dist_util.erl

```diff
@@ -695,8 +695,14 @@ send_status(#hs_data{socket = Socket, other_node = Node,
 	_ -> 
 	    true
     end.
-    
-    
+
+%% Constant time comparison between two binaries of the same size.
+secure_compare(B1, B2) ->
+    secure_compare(B1, B2, 0) =:= 0.
+secure_compare(<<H1, T1/binary>>, <<H2, T2/binary>>, Acc) ->
+    secure_compare(T1, T2, Acc bor (H1 bxor H2));
+secure_compare(<<>>, <<>>, Acc) ->
+    Acc.
```

Comment:
I disagree for two reasons: 1. the function header is clearly documented it is for comparing binaries of the same size. 2. the protocol currently specifies the challenge must have length of 16 bytes. This is verified when parsing the binary packets. If the OTP team thinks this should be more future proof, I am glad to change, but I won't make it more complex unless there is a need to (or you are welcome to send another PR later).